app/controllers/proposals_controller.rb

```diff
@@ -50,7 +50,7 @@ def cancel
   end
 
   def approve
-    approval = proposal.existing_approval_for(current_user)
+    approval = proposal.existing_step_for(current_user)
```

Comment:
we do not know if is an approval type, but both Approval and Purchase steps hit this `approve` method. A completed purchase step has a state of `approved` (at least I think this is the case...paging @yozlet )